src/SegmentationCameraSensor.cc

```diff
@@ -467,16 +482,29 @@ bool SegmentationCameraSensorPrivate::SaveImage(
       return false;
   }
 
-  std::string filename = this->saveImagePrefix +
+  std::string coloredName = this->saveImagePrefix + "labels_" +
                          std::to_string(this->saveImageCounter) + ".png";
+  std::string labelsName = this->saveImagePrefix + "colored_" +
```

Comment:
```suggestion
  std::string coloredName = this->saveImagePrefix + "colored_" +
                         std::to_string(this->saveImageCounter) + ".png";
  std::string labelsName = this->saveImagePrefix + "labels_" +
```
it looks like the strings were assigned to the wrong variable names, probably because this method was called with parameters in reverse order as I mentioned in my previous comment